sdk/src/account.rs

```diff
@@ -21,112 +21,455 @@ pub struct Account {
     pub rent_epoch: Epoch,
 }
 
+/// An Account with data that is stored on chain
+/// This will become a new in-memory representation of the 'Account' struct data.
+/// The existing 'Account' structure cannot easily change due to downstream projects.
+/// This struct will shortly rely on something like the AnAccount trait for access to the fields.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Default, AbiExample)]
+pub struct AccountSharedData {
+    /// lamports in the account
+    pub lamports: u64,
+    /// data held in this account
+    #[serde(with = "serde_bytes")]
+    pub data: Vec<u8>, // will be: Arc<Vec<u8>>,
```

Comment:
If this data is shared betwen threads (central cache?) do we also need to wrap with a mutex or are you thinking copy on write?

Reply:
> If this data is shared betwen threads (central cache?) do we also need to wrap with a mutex or are you thinking copy on write?

copy on write. The semantics today are multiple identical copies of the data, but any copy can be modified. This PR is moving towards multiple references to shared data (for performance) where if any reference wants to modify its data, it gets a new copy of the data and the other refers retain references to the unchanged data.